src/main/kotlin/com/avail/descriptor/pojos/PojoDescriptor.kt

```diff
@@ -0,0 +1,210 @@
+import com.avail.descriptor.pojos.PojoDescriptor.ObjectSlots.RAW_POJO
+import com.avail.descriptor.pojos.RawPojoDescriptor.Companion.rawNullPojo
+import com.avail.descriptor.representation.A_BasicObject
+import com.avail.descriptor.representation.AvailObject
+import com.avail.descriptor.representation.Mutability
+import com.avail.descriptor.representation.ObjectSlotsEnum
+import com.avail.descriptor.types.A_Type
+import com.avail.descriptor.types.BottomPojoTypeDescriptor.pojoBottom
+import com.avail.descriptor.types.PojoTypeDescriptor
+import com.avail.descriptor.types.TypeTag
+import com.avail.utility.json.JSONWriter
+import java.util.*
+
+/**
+ * A `PojoDescriptor` describes a plain-old Java object (pojo) that is
+ * accessible to an Avail programmer as an [AvailObject]. An Avail pojo
+ * comprises a [raw&#32;pojo][RawPojoDescriptor] and a
+ * [pojo&#32;type][PojoTypeDescriptor] that describes the pojo contextually.
+ *
+ * @constructor
+ *
+ * @param mutability
+ *   The [mutability][Mutability] of the new descriptor.
+ *
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ */
+class PojoDescriptor private constructor(
+	mutability: Mutability
+) : Descriptor(mutability, TypeTag.POJO_TAG, ObjectSlots::class.java, null) {
+	/** The layout of the object slots.  */
+	enum class ObjectSlots : ObjectSlotsEnum {
+		/** A [raw&#32;pojo][RawPojoDescriptor]. */
+		RAW_POJO,
+
+		/** The [kind][PojoTypeDescriptor] of the pojo. */
+		KIND
+	}
+
+	@AvailMethod
+	override fun o_Equals(self: AvailObject, another: A_BasicObject): Boolean =
+		another.equalsPojo(self)
+
+	@AvailMethod
+	override fun o_EqualsPojo(self: AvailObject, aPojo: AvailObject): Boolean =
+		when {
+			self.sameAddressAs(aPojo) -> true
+			!self.slot(RAW_POJO).equals(aPojo.slot(RAW_POJO)) -> false
+			else -> {
+				if (!isShared) {
+					self.becomeIndirectionTo(aPojo.makeImmutable())
+				}
+				else if (!aPojo.descriptor().isShared) {
+					aPojo.becomeIndirectionTo(self.makeImmutable())
+				}
+				true
+			}
+		}
+
+	@AvailMethod
+	override fun o_Hash(self: AvailObject): Int {
+		var hash = self.slot(RAW_POJO).hash() xor 0x749101DD
+		hash *= AvailObject.multiplier
+		hash += self.slot(KIND).hash()
+		return hash
+	}
+
+	@AvailMethod
+	override fun o_IsPojo(self: AvailObject): Boolean = true
+
+	@AvailMethod
+	override fun o_Kind(self: AvailObject): A_Type = self.slot(KIND)
+
+	override fun o_MarshalToJava(
+		self: AvailObject,
+		ignoredClassHint: Class<*>?
+	): Any? {
+		return self.slot(RAW_POJO).javaObject<Any>()
+	}
+
+	@AvailMethod
+	override fun o_RawPojo(self: AvailObject): AvailObject = self.slot(RAW_POJO)
+
+	@AvailMethod
+	override fun <T> o_JavaObject(self: AvailObject): T? =
+		self.slot(RAW_POJO).javaObject()
+
+	override fun o_ShowValueInNameForDebugger(self: AvailObject): Boolean =
+		false
+
+	override fun o_WriteTo(self: AvailObject, writer: JSONWriter) {
+		writer.startObject()
+		writer.write("kind")
+		writer.write("pojo")
```

Comment:
When using our JSON library, I like to make the key-value pair as one statement as the `write` functions are actually an expression that evaluate to `JSONWriter`. So something like:
`writer.write("kind").write("pojo")`
Doing this makes the intent of the author more clear as to the desired resulting JSON.

Reply:
Ok, I think you'll like this change.  We can update all the other implementations at a later date.